### PR TITLE
Andrius Isin - London 10 - Tv-show-dom-project

### PR DESCRIPTION
**Pull request by AndriusIsin**

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,12 +1,125 @@
...
+const searchElement = document.getElementById("search-input");
+const numberOfEpisodes = document.getElementById("episodes-number");
+const rootElem = document.getElementById("root");
+const allEpisodes = getAllEpisodes();
+const shows = getAllShows();
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  fetch("https://api.tvmaze.com/shows/82/episodes")
```
Comment (llamojha) [inline script.js:10]: Excellent implementation of this. I recommend creating another js file with the getAllEpisodes() function that contains the whole JSON file for that call .Mainly so you can test it 'offline' without having to do a call to the tvmaze API.

File: script.js [hunk 2]
```diff
@@ -1,12 +1,125 @@
...
+}
+function addOptionsShow() {
+  let showTitles = shows.map((show) => show.name).sort();
+  showTitles.forEach((show) => {
+    const optionElement = document.createElement("option");
+    optionElement.innerHTML = show;
+    selectElementShow.appendChild(optionElement);
+  });
+}
+addOptionsShow();
```
Comment (llamojha) [inline script.js:81]: This call to addOptionsShow() function is in between functions. Move it somewhere so the code has more structure

File: index.html [hunk 3]
```diff
@@ -3,18 +3,41 @@
...
+          </p>
+         
+
+    </section>
+    <section id="root">
+
+    </section>
+    <footer>
+      <p>
+        Developed by Andrius Isin
```
[the remark this answers is not in the document]
Comment (AndriusIsin) [reply index.html:32]: Thank you for your review Alvaro!

File: script.js [hunk 4]
```diff
@@ -1,12 +1,206 @@
...
+      .catch((error) => console.log(error));
+  });
 }
 
+//Level 500
+function showAllShows(shows) {
+  selectElementShow.style.display = "inline";
+  selectElement.style.display = "none";
+  shows.forEach((show) => {
+    const showContainer = document.createElement("div");
```
Comment (illicitonion) [inline script.js:147]: There are a lot of calls here that do the same thing but slightly differently, which are a bit hard to follow; lots of calls to `document.createElement` then `element.classList =` then `element.innerText =` then `parent.appendChild(element)` - can you think of a useful function to extract to make it more clear what's happening each time?

**Review by illicitonion — COMMENTED** [review]
This code generally looks really solid - I like how you've pulled out lots of useful functions with clear names, and generally given variables a lot of good names. 

From using your site in the UI, it looks like when the show-selection stuff got added, the episode-selection stuff may have broken? Do you know what went wrong there?

File: script.js [hunk 5]
```diff
@@ -1,12 +1,208 @@
...
+  // fetchEpisodes();
+}
+// level 100
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  selectElementShow.style.display = "inline";
+  selectElement.style.display = "inline";
+  rootElemShow.textContent = "";
+  episodeList.forEach((episode) => addEpisode(episode));
```
Comment (illicitonion) [inline script.js:28]: Is there an array function you could use instead of `forEach` here?

Can you look for replacements for your other calls to `forEach` too?
